pkg/operator/ceph/config/store.go

```diff
@@ -0,0 +1,151 @@
+func (s *Store) CreateOrUpdate(clusterInfo *cephconfig.ClusterInfo) error {
+	c := DefaultCentralizedConfigs(s.context.LogLevel)
+
+	/* TODO: in the future merge config overrides from the cluster CRD */
+
+	t, err := c.FileText()
+	if err != nil {
+		return fmt.Errorf("failed to store the Ceph config file. %+v", err)
+	}
+	if err := s.configMapStore.SetValue(storeName, confFileName, t); err != nil {
+		return fmt.Errorf("failed to store the Ceph config file. %+v", err)
+	}
+	logger.Infof("Generated config file:\n%s", t)
+
+	hosts := make([]string, len(clusterInfo.Monitors))
+	members := make([]string, len(clusterInfo.Monitors))
+	i := 0
+	for _, m := range clusterInfo.Monitors {
+		hosts[i] = m.Endpoint
+		members[i] = m.Name
+		i++
+	}
+	// mon_host is string list of mon endpoints separated in form "IP:PORT,IP:PORT,etc."
+	err = s.configMapStore.SetValue(storeName, "mon_host", strings.Join(hosts, ","))
+	if err != nil {
+		return fmt.Errorf("failed to store the Ceph mon_host info. %+v", err)
+	}
+
+	// mon_initial_members is string list of mon ids separated in form "ID,ID,ID,etc."
+	err = s.configMapStore.SetValue(storeName, "mon_initial_members", strings.Join(members, ","))
+	if err != nil {
+		return fmt.Errorf("failed to store the Ceph mon_initial_members info. %+v", err)
+	}
+
+	/*
+		TODO:
+		if Luminous {
+			store config in config map as above
+		} else if Mimic + {
+			set values with `ceph config assimilage-conf t`
+		}
+	*/
+
+	return nil
+}
+
+// StoredFileVolume returns a pod volume sourced from the stored config file.
+func StoredFileVolume() v1.Volume {
+	// TL;DR: mount the configmap's "ceph.conf" to a file called "ceph.conf" with 0400 permissions
+	mode := int32(0400) // security: only allow the owner to read and no one to write
+	return v1.Volume{
+		Name: configVolumeName,
+		VolumeSource: v1.VolumeSource{
+			ConfigMap: &v1.ConfigMapVolumeSource{LocalObjectReference: v1.LocalObjectReference{
+				Name: storeName},
+				Items: []v1.KeyToPath{
+					{Key: confFileName, Path: confFileName, Mode: &mode},
+				}}}}
+}
+
+// StoredFileVolumeMount returns a container volume mount that mounts the stored config from
+// StoredFileVolume into the container at `/etc/ceph/ceph.conf`.
+func StoredFileVolumeMount() v1.VolumeMount {
+	// configmap's "ceph.conf" to "/etc/ceph/ceph.conf"
+	return v1.VolumeMount{
+		Name:      storeName,
+		ReadOnly:  true, // should be no reason to write to the config in pods, so enforce this
+		MountPath: "/etc/ceph",
+	}
+}
+
+// StoredMonHostEnvVars returns a container environment variable defined by the most updated stored
+// "mon_host" and "mon_initial_members" information.
+func StoredMonHostEnvVars() []v1.EnvVar {
+	return []v1.EnvVar{
+		{Name: "ROOK_CEPH_MON_HOST",
+			ValueFrom: &v1.EnvVarSource{
+				ConfigMapKeyRef: &v1.ConfigMapKeySelector{LocalObjectReference: v1.LocalObjectReference{
+					Name: storeName},
+					Key: "mon_host"}}},
```

Comment:
could you define consts for all the setting keys we're using? If a string is used multiple places besides tests at least that's what i'd recommend

-